tests/phpunit/tests/option/option.php

```diff
@@ -359,16 +359,85 @@ public function test_option_autoloading( $name, $autoload_value, $expected ) {
 	 */
 	public function data_option_autoloading() {
 		return array(
-			array( 'autoload_yes', 'yes', 'yes' ),
-			array( 'autoload_true', true, 'yes' ),
-			array( 'autoload_string', 'foo', 'yes' ),
-			array( 'autoload_int', 123456, 'yes' ),
-			array( 'autoload_array', array(), 'yes' ),
-			array( 'autoload_no', 'no', 'no' ),
-			array( 'autoload_false', false, 'no' ),
+			array( 'autoload_yes', 'yes', 'on' ),
+			array( 'autoload_true', true, 'on' ),
+			array( 'autoload_string', 'foo', 'auto' ),
+			array( 'autoload_int', 123456, 'auto' ),
+			array( 'autoload_array', array(), 'auto' ),
+			array( 'autoload_no', 'no', 'off' ),
+			array( 'autoload_false', false, 'off' ),
+			array( 'autoload_null', null, 'auto' ),
 		);
 	}
 
+	/**
+	 *
+	 * @ticket 42441
+	 * @dataProvider data_option_autoloading_large_option
+	 *
+	 * @covers ::update_option
+	 */
+	public function test_update_option_autoloading_large_option( $autoload, $expected ) {
+		global $wpdb;
+		$name = 'foo';
+		add_option( $name, 'bar' );
+		add_filter( 'wp_max_autoloaded_option_size', array( $this, 'filter_max_option_size' ) );
+		$value = file( DIR_TESTDATA . '/formatting/entities.txt' );
+		$added = update_option( $name, $value, $autoload );
+		$this->assertTrue( $added );
+
+		$actual = $wpdb->get_row( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $name ) );
+		$this->assertSame( $expected, $actual->autoload );
+	}
+
+	public function data_option_autoloading_large_option() {
+		return array(
+			'yes'   => array(
+				'autoload' => 'yes',
+				'expected' => 'on',
+			),
+			'true'  => array(
+				'autoload' => true,
+				'expected' => 'on',
+			),
+			'no'    => array(
+				'autoload' => 'no',
+				'expected' => 'off',
+			),
+			'false' => array(
+				'autoload' => false,
+				'expected' => 'off',
+			),
+			'null'  => array(
+				'autoload' => null,
+				'expected' => 'auto-off',
+			),
+		);
+	}
+
+	public function filter_max_option_size( $current ) {
+		return 1000;
+	}
+
+	/**
+	 *
+	 * @ticket 42441
+	 *
+	 * @covers ::update_option
+	 */
+	public function test_update_option_autoloading_large_option_no() {
+		global $wpdb;
+		$name = 'foo';
+		add_option( $name, 'bar', '', 'no' );
+		add_filter( 'wp_max_autoloaded_option_size', array( $this, 'filter_max_option_size' ) );
+		$value = file( DIR_TESTDATA . '/formatting/entities.txt' );
+		$added = update_option( $name, $value );
+		$this->assertTrue( $added );
+
+		$actual = $wpdb->get_row( $wpdb->prepare( "SELECT autoload FROM $wpdb->options WHERE option_name = %s LIMIT 1", $name ) );
+		$this->assertSame( 'off', $actual->autoload );
+	}
```

Comment:
What is the purpose of this test? It looks like it's almost doing the same as the `test_update_option_autoloading_large_option` test above.

I think it would make more sense to add another test where the file is actually _smaller_ than the "max option size", and then assert that the value is `'auto'`.

Reply:
done